packages/common/src/components/CondaPkgPanel.tsx

```diff
@@ -468,58 +494,84 @@ export class CondaPkgPanel extends React.Component<
    * @return {Promise<void>}
    */
   async onPkgBottomHit(): Promise<void> {
-    if (!this.state.isLoading) {
-      this.setState({
-        isLoading: true
-      });
-      const packages = await this._model.loadMorePackages?.();
+    if (this.state.isLoading) {
+      return;
+    }
+
+    this.setState({
+      isLoading: true
+    });
+    if (this.state.searchTerm) {
+      console.log(
+        'onPkgBottomHit, loading search packages, searchTerm',
+        this.state.searchTerm
+      );
+      const searchMatchPackages = await this._model.searchPackages?.(
+        this.state.searchTerm
+      );
+      if (searchMatchPackages !== undefined) {
+        this.setState({ searchMatchPackages });
+      }
+    } else {
+      console.log('onPkgBottomHit, loading installed packages');
+      const packages = await this._model.loadInstalledPackages?.();
       if (packages !== undefined) {
         this.setState({ packages });
       }
-      this.setState({
-        isLoading: false
-      });
     }
+
+    this.setState({
+      isLoading: false
+    });
   }
 
   render(): JSX.Element {
-    let filteredPkgs: Conda.IPackage[] = [];
-    if (this.state.activeFilter === PkgFilters.All) {
-      filteredPkgs = this.state.packages;
-    } else if (this.state.activeFilter === PkgFilters.Installed) {
-      filteredPkgs = this.state.packages.filter(pkg => pkg.version_installed);
-    } else if (this.state.activeFilter === PkgFilters.Available) {
-      filteredPkgs = this.state.packages.filter(pkg => !pkg.version_installed);
-    } else if (this.state.activeFilter === PkgFilters.Updatable) {
-      filteredPkgs = this.state.packages.filter(pkg => pkg.updatable);
-    } else if (this.state.activeFilter === PkgFilters.Selected) {
-      filteredPkgs = this.state.packages.filter(
-        pkg => this.state.selected.indexOf(pkg) >= 0
-      );
+    switch (this.state.buildStatus) {
+      case 'COMPLETED':
+        break;
+      case 'FAILED':
+        return <div>The build for this environment failed.</div>;
+      case 'QUEUED':
+      case 'BUILDING':
+      default:
+        return <div>The build for this environment is pending.</div>;
```

Comment:
It's best if the end user doesn't see this markup, so I've made certain operations hang until the build is done (for example, creating/cloning an environment)